File: 2024/Python/Challenge#6.py
from typing import List


def in_box(box: List[str]) -> bool:
    for i in range(1, len(box) - 1):
        if '*' in box[i][1:-1]:
            return True                
    return False


result = in_box([
  "###",
  "#*#",
  "###"
])  # ➞ true
print(result)

result = in_box([
  "####",
  "#* #",
  "#  #",
  "####"
])  # ➞ true
print(result)

result = in_box([
  "#####",
  "#   #",
  "#  #*",
  "#####"
])  # ➞ false
print(result)

result = in_box([
  "#####",
  "#   #",
  "#   #",
  "#   #",
  "#####"
])  # ➞ false
print(result)

File: 2024/TypeScript/Challenge#6.ts

function inBox(box: string[]): boolean {
    for (let i = 1; i < box.length - 1; i++) {
        if (box[i].substring(1, box[i].length - 1).includes('*')) {
            return true;
        }
    }
    return false
}


const bool = inBox([
    "###",
    "#*#",
    "###"
]) // ➞ true
console.log(bool);

const bool2 = inBox([
    "####",
    "#* #",
    "#  #",
    "####"
]) // ➞ true
console.log(bool2);

const bool3 = inBox([
    "#####",
    "#   #",
    "#  #*",
    "#####"
]) // ➞ false
console.log(bool3);

const bool4 = inBox([
    "#####",
    "#   #",
    "#   #",
    "#   #",
    "#####"
]) // ➞ false
console.log(bool4);
